vault/plugin_catalog.go

```diff
@@ -207,6 +207,12 @@ func (c *PluginCatalog) cleanupExternalPlugin(name, id string) error {
 	c.logger.Debug("removed plugin client connection", "id", id)
 
 	if !extPlugin.multiplexingSupport {
+		clientConn, err := pc.client.Client()
+		if err != nil {
+			return err
```

Comment:
We might want to make this best-effort and simply log the error, but continue with the rest of the cleanup.